src/hotspot/share/runtime/mutex.cpp

```diff
@@ -463,7 +446,11 @@ void Mutex::set_owner_implementation(Thread *new_owner) {
     new_owner->_owned_locks = this;
 
     // NSV implied with locking allow_vm_block flag.
-    no_safepoint_verifier(new_owner, true);
+    // The tty_lock is special because it is released for the safepoint by
+    // the safepoint mechanism.
+    if (new_owner->is_Java_thread() && _allow_vm_block && this != tty_lock) {
+      new_owner->as_Java_thread()->inc_no_safepoint_count();
+    }
```

Comment:
I didn't notice this on first pass.  I like the change because it was an awkward function anyway because of the enable/disable parameter.